frontend/src/pages/projects/notebook/useRelatedNotebooks.ts

```diff
@@ -25,6 +25,18 @@ const useRelatedNotebooks = (
   } = React.useContext(ProjectDetailsContext);
 
   const connectedNotebooks = React.useMemo(() => {
+    if (context === ConnectedNotebookContext.POSSIBLE_DATA_CONNECTION) {
```

Comment:
I don't like lol -- this belongs in the switch statement... maybe we inline the escape early for no resource name for the other calls? 🤔 

I want to keep the switch statement full for each case.